pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+    pub fn RewardConfigOnEmpty() -> RewardDistributionConfig {
+        RewardDistributionConfig::default()
+    }
+
+    /// An active list of configuration parameters used to calculate reward distribution portions.
+    #[pallet::storage]
+    #[pallet::getter(fn reward_config)]
+    pub type ActiveRewardDistributionConfig<T> =
+        StorageValue<_, RewardDistributionConfig, ValueQuery, RewardConfigOnEmpty>;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub (super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        Staked { who: T::AccountId, creator: SpaceId, era: EraIndex, amount: BalanceOf<T> },
+        Unstaked { who: T::AccountId, creator: SpaceId, era: EraIndex, amount: BalanceOf<T> },
+        RewardsClaimed { who: T::AccountId, amount: BalanceOf<T> },
+        WithdrawnFromClaimed { who: T::AccountId, amount: BalanceOf<T> },
+        WithdrawnFromUnregistered { who: T::AccountId, amount: BalanceOf<T> },
+        AnnualInflationSet { value: Perbill },
+        RewardsCalculated { total_rewards_amount: BalanceOf<T> },
+        CreatorRegistered { who: T::AccountId, space_id: SpaceId },
+        CreatorUnregistered { who: T::AccountId, space_id: SpaceId },
+        CreatorUnregisteredWithSlash { who: T::AccountId, space_id: SpaceId, amount: BalanceOf<T> },
+        NewCreatorStakingEra { number: EraIndex },
+        MaintenanceModeSet { status: bool },
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Pallet is disabled.
+        PalletIsDisabled,
+        AlreadyUsedCreatorSpace,
+        NotRegisteredCreator,
+        NotACreator,
+        CannotStakeZero,
+        CannotUnstakeZero,
+        MaxNumberOfStakersExceeded,
+        UnexpectedStakeInfoEra,
+        TooManyEraStakeValues,
+        InsufficientValue,
+        NotStakedCreator,
+        TooManyUnlockingChunks,
+        NothingToWithdraw,
+        NotUnregisteredCreator,
+        UnclaimedRewardsRemaining,
+        EraOutOfBounds,
+        UnknownEraReward,
+        AlreadyClaimedInThisEra,
+        MaintenanceModeNotChanged,
+        RewardDistributionConfigInconsistent,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_initialize(now: BlockNumberFor<T>) -> Weight {
+            // As long as pallet is disabled, we shouldn't allow any storage modifications.
+            // This means we might prolong an era but it's acceptable.
+            // Runtime upgrade should be timed so we ensure that we complete it before
+            // a new era is triggered. This code is just a safety net to ensure nothing is broken
+            // if we fail to do that.
+            if PalletDisabled::<T>::get() {
+                return T::DbWeight::get().reads(1);
+            }
+
+            let force_new_era = Self::force_era().eq(&Forcing::ForceNew);
+            let previous_era = Self::current_era();
+            let next_era_starting_block = Self::next_era_starting_block();
```

Comment:
How about combining these tree storages into one struct. Then we will do 1 read + 1 write in this `on_initialize` fn.